apis/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -292,6 +292,11 @@ type AdmissionControllerFeatureConfig struct {
 	// ServiceName corresponds to the webhook service name.
 	// +optional
 	ServiceName *string `json:"serviceName,omitempty"`
+
+	// configInjectionMode corresponds to the kind of configuration to be injected.
+	// It can be "hostip", "service" or "socket".
+	// +optional
+	ConfigInjectionMode *string `json:"configInjectionMode,omitempty"`
```

Comment:
Since we also plan to support other use cases in the admission controller. We should be more precise on this configuration option

```suggestion

	// agentCommunicationMode corresponds to the mode use by the Datadog applicative libraries to communicate with the Agent. The mode will define also how this communication will be made.
	// It can be "hostip", "service" or "socket".
	// +optional
	AgentCommunicationMode *string `json:"agentCommunicationMode,omitempty"`
```

Reply:
makes sense!